reactor-core/src/main/java/reactor/core/publisher/Processors.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2011-Present VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import java.time.Duration;
+import java.util.Queue;
+import java.util.function.Consumer;
+
+import reactor.core.Disposable;
+import reactor.core.scheduler.Scheduler;
+import reactor.util.annotation.Nullable;
+
+/**
+ * @author Simon Baslé
+ */
+public final class Processors {
+
+
+	@SuppressWarnings("deprecation")
+	public static final <T> FluxProcessor<T, T> unicast() {
```

Comment:
FYI one of the reasons why I was choosing concrete type over `FluxProcessor` is because having to specify the type argument twice was a bit cumbersome.

As this is a new API, perhaps we can do something about it? especially given that now it is *always* `<T, T>`

Reply:
the number of new interfaces / classes was a concern in the original proposal back then, but there's probably little way around the current split into 4 sink interfaces (plus 2 additional utility classes).

we could improve that `<T,T>` situation but it means adding `FluxIdentityProcessor<T>` at least... Maybe `Processors` can be turned into both interface and factory.

 It raises another issue, which is the misalignment between `FluxProcessor<A,B>` and `MonoProcessor<A>` (mono processor has a single concrete implementation, and thus doesn't help with representing processing stages). That'll be a concern for either M2 or 3.5 now :/

Reply:
actually it cannot be melded with `Processors`, the `FluxProcessor` being an abstract class (due to the need to extend `Flux`)... furthermore `replay` factory method on `Processors` would clash with `Flux#replay()`, and the factory methods would be lost in the noise of `Flux` API.

Instead I went ahead and added an abstract intermediate `FluxIdentityProcessor<T> extends FluxProcessor<T, T>`